lib/Twig/Node/Expression/Call.php

```diff
@@ -287,6 +300,93 @@ private function reflectCallable($callable)
 
         return $this->reflector = array($r, $callable);
     }
+
+    public function compileInline(Twig_Compiler $compiler, $callable)
+    {
+        $args = array();
+
+        if ($this->hasAttribute('needs_environment') && $this->getAttribute('needs_environment')) {
+            $args[] = $compiler->getEnvironment();
+        }
+
+        if ($this->hasAttribute('needs_context') && $this->getAttribute('needs_context')) {
+            $args[] = array(); // @TODO: Is it possible to get the context from here?
+        }
+
+        if ($this->hasAttribute('arguments')) {
+            foreach ($this->getAttribute('arguments') as $argument) {
+                $args[] = sprintf('"%s"', addcslashes($argument, "\0\t\"\$\\"));
```

Comment:
that looks wrong to me

Reply:
This just mimics the `$compiler->string()` function. It's required to escape the returned values (escaping quotes is especially necessary here)

Reply:
but that's totally wrong. You are evaluating code here. So you want a string (which you already have), not a string containing the source code for a PHP string (you won't run this through `eval`)

Reply:
I'm quite sure you haven't tried this code at all.

Reply:
I did try the code, I wouldn't submit a PR for something that I haven't tested.

The initial version of my changes caused errors in the compiled template if any string is returned with quotes. Since the raw value was added to the compiled template, you would end up with `echo twig_escape_filter($this->env, one "two" three, "html", null, true);` instead of ` `echo twig_escape_filter($this->env, "one \"two\" three", "html", null, true);`. So this was necessary.
But after the refactor I ended up using `$compiler->string` (instead of `$compiler->raw`) which now handles the escaping of the values. So this part is not needed anymore, but it was necessary with the initial patch before I switched from `->raw()` to `->string()`

Reply:
and now, it is breaking things